pkg/server/server.go

```diff
@@ -0,0 +1,301 @@
+package server
+
+import (
+	"context"
+	"crypto/tls"
+	"crypto/x509"
+	"errors"
+	"fmt"
+	"github.com/kcp-dev/kcp/pkg/etcd"
+	"github.com/kcp-dev/kcp/pkg/reconciler/apiresource"
+	"github.com/kcp-dev/kcp/pkg/reconciler/cluster"
+	clientv3 "go.etcd.io/etcd/client/v3"
+	"io/ioutil"
+	genericapiserver "k8s.io/apiserver/pkg/server"
+	"k8s.io/apiserver/pkg/storage/storagebackend"
+	"k8s.io/client-go/tools/clientcmd"
+	clientcmdapi "k8s.io/client-go/tools/clientcmd/api"
+	"k8s.io/kubernetes/pkg/genericcontrolplane"
+	"k8s.io/kubernetes/pkg/genericcontrolplane/clientutils"
+	"k8s.io/kubernetes/pkg/genericcontrolplane/options"
+	"net"
+	"net/url"
+	"os"
+	"path/filepath"
+	"strconv"
+)
+
+// Server manages the configuration and kcp api-server. It allows callers to easily use kcp
+// as a library rather than as a single binary. Using its constructor function, you can easily
+// setup a new api-server and start it:
+//
+//	srv := server.NewServer(server.DefaultConfig())
+//  srv.Run(ctx)
+//
+// You may optionally provide PostStartHookFunc and PreShutdownHookFunc hooks before starting
+// the server that should be passed to the api-server itself. These hooks have access to a
+// restclient.Config which allows you to easily create a client.
+//
+//  srv.AddPostStartHook("my-hook", func(context genericapiserver.PostStartHookContext) error {
+//		client := clientset.NewForConfigOrDie(context.LoopbackClientConfig)
+//	})
```

Comment:
Are there some formatting problems here ?

Reply:
Interesting, it looks fine in my IDE. I'll commit a few things and see if I can get it right.

Reply:
Alright, we should be good now

https://github.com/kcp-dev/kcp/pull/153/files#diff-104a11712684e520f0affa95e6a053faeb713306a10f69ee99b9729cc9dd1996R35-R41